### ISSUE TITLE
Return type of unordered_map erase methods should be iterator not void

### ISSUE THREAD

**Issue by carlosdiez** [cython/cython#1484]
The return type of unordered_map erase methods should be `iterator` not `void`, like in `vector`.

https://github.com/cython/cython/blob/master/Cython/Includes/libcpp/unordered_map.pxd#L45

C++ reference: http://en.cppreference.com/w/cpp/container/unordered_map/erase

**Comment by robertwb** [comment]
Thanks. We'd welcome a PR with a test :).

**Comment by carlosdiez** [comment]
Ok, working on that :)

**Comment by carlosdiez** [comment]
Created a PR for this https://github.com/cython/cython/pull/1490
I think it would be good to add more tests for `erase` method on other data structures.

Also, because `unordered_map` is from C++11, I don't know if it will compile on all platforms.

I added `-std=c++11` to `CXXFLAGS`

```python
(dev) carlos@macbookair:~/code/cython$ git diff HEAD
diff --git a/runtests.py b/runtests.py
index 0e0abf5..01f44a2 100755
--- a/runtests.py
+++ b/runtests.py
@@ -423,6 +423,7 @@ class build_ext(_build_ext):
                 compiler_obj = self.compiler
             if ext.language == 'c++':
                 compiler_obj.compiler_so.remove('-Wstrict-prototypes')
+                compiler_obj.compiler_so.append('-std=c++11')
             if CCACHE:
                 compiler_obj.compiler_so = CCACHE + compiler_obj.compiler_so
             if getattr(ext, 'openmp', None) and compiler_obj.compiler_type == 'msvc':
(dev) carlos@macbookair:~/code/cython$ 
```

**Comment by scoder** [comment]
Fixed in #1933.
